scripts/merge-script/pull-request.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as Octokit from '@octokit/rest';
+import {PullRequestFailure} from './failures';
+import {GitClient} from './git';
+import {PullRequestMergeTask} from './index';
+import {matchesPattern} from './string-pattern';
+import {getBranchesFromTargetLabel, getTargetLabelFromPullRequest} from './target-label';
+
+/** Interface that describes a pull request. */
+export interface PullRequest {
+  /** Number of the pull request. */
+  prNumber: number;
+  /** Title of the pull request. */
+  title: string;
+  /** Labels applied to the pull request. */
+  labels: string[];
+  /** List of branches this PR should be merged into. */
+  targetBranches: string[];
+  /** Branch that the PR targets in the Github UI. */
+  githubTargetBranch: string;
+  /** Count of commits in this pull request. */
+  commitCount: number;
+  /** Optional SHA that this pull request needs to be based on. */
+  requiredBaseSha?: string;
+  /** Whether the pull request commit message fixup. */
+  needsCommitMessageFixup: boolean;
+}
+
+/**
+ * Loads and validates the specified pull request against the given configuration.
+ * If the pull requests fails, a pull request failure is returned.
+ */
+export async function loadAndValidatePullRequest(
+    {git, config}: PullRequestMergeTask, prNumber: number,
+    ignoreNonFatalFailures = false): Promise<PullRequest|PullRequestFailure> {
+  const prData = await fetchPullRequestFromGithub(git, prNumber);
+
+  if (prData === null) {
+    return PullRequestFailure.notFound();
+  }
+
+  const labels = prData.labels.map(l => l.name);
+
+  if (!labels.some(name => matchesPattern(name, config.mergeReadyLabel))) {
+    return PullRequestFailure.notMergeReady();
+  } else if (!labels.some(name => matchesPattern(name, config.claSignedLabel))) {
+    return PullRequestFailure.claUnsigned();
+  }
+
+  const {data: {state}} =
+      await git.api.repos.getCombinedStatusForRef({...git.repoParams, ref: prData.head.sha});
+
+  const targetLabel = getTargetLabelFromPullRequest(config, labels);
+  if (targetLabel === null) {
+    return PullRequestFailure.noTargetLabel();
+  }
```

Comment:
Can we move the data request to after the target label check?  Just so we don't request the data and then fail before we even use it.
```suggestion
  const targetLabel = getTargetLabelFromPullRequest(config, labels);
  if (targetLabel === null) {
    return PullRequestFailure.noTargetLabel();
  }
  
  const {data: {state}} =
      await git.api.repos.getCombinedStatusForRef({...git.repoParams, ref: prData.head.sha});
```

Reply:
Yeah, fixed!